packages/flutter_tools/lib/src/base/config.dart

```diff
@@ -97,4 +108,12 @@ class Config {
       : 'HOME';
     return platform.environment[envKey] ?? '.';
   }
+
+  static String _userConfigPath(Platform platform, FileSystem filesystem) {
+    if (platform.operatingSystem == 'windows') {
```

Comment:
This seems wrong? Why no `if (!platform.isLinux) { ... }`

Reply:
You are right. I do however think `$XDG_CONFIG_HOME` should be the default on macOS as well, so it would be `if (!platform.isLinux && !platform.isMacOS)`.